lib/gatsby.ts

```diff
@@ -0,0 +1,47 @@
+import { exec } from 'child_process';
+
+
+/**
+ * Check if gatsby-cli is installed globally.
+ * @return {Promise<boolean>}
+ */
+function checkForGatsby(): Promise<boolean> {
+  return new Promise<boolean>((resolve, reject) => {
+    exec('npm list -g gatsby-cli --json', (error, stdout, stderr) => {
+      if (error && stdout === '{}') reject(error);
+
+      let resultObj: any = JSON.parse(stdout);
+
+      if (Object.keys(resultObj).length === 0 && resultObj.constructor === Object) resolve(false);
+      else resolve(true);
+    });
+  });
+}
+
+/**
+ * Checks if gatsby is already installed globally and if not installs it.
+ * @return {Promise<void>}
+ */
+export function installGatsby(): Promise<void> {
+  return new Promise<void>((resolve, reject) => {
+    checkForGatsby().then(
+      (value) => {
+        if (value) {
+          console.info('gatsby-cli is already installed');
+          resolve();
+        } else {
+          console.info('gatsby-cli not installed. Installing...');
+          exec('npm install -g gatsby-cli --quiet', (error, stdout, stderr) => {
+            if (error) reject(error);
+
+            console.log(`stdout: ${stdout}`);
+            console.error(`stderr: ${stderr}`);
+            console.info('Finished installation of gatsby-cli');
```

Comment:
Unsure if there is any value to logging this stuff in the console here. Any idea how this behaves when the app is built?

On the other hand, it might be a good idea to save logs only if install fails. Dumping it into a file could maybe provide some value if for some reason this fails in production and user want's to find out what went wrong?

What do you think? 🤔

Reply:
The way it works now is that you can see all of the output in the dev tools console after the installation finishes. I would keep some logging capabilities because:
- as you have said, if some problems occur it will help with debugging
- gatsby-cli installation is a one-time thing and there isn't a lot of info to log, so the performance and log size cost are negligible
That being said, logging to a file is better, so I'll change it.

Reply:
If we run this in main thread (either by sending signal via IPC or using `createActionMain` helper to create Redux action) the output will not show up in devtools as it's only run in main process and it would output to that console. So the question would be if it opens up that console in production.

We might also want to later attach this function to some state that front-end team could use to either show that it's installing but I'll consult with them about this.